zipline/data/us_equity_loader.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2016 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from numpy import dtype, around
+
+from six import iteritems
+
+from zipline.pipeline.data.equity_pricing import USEquityPricing
+from zipline.lib._float64window import AdjustedArrayWindow as Float64Window
+from zipline.lib._int64window import AdjustedArrayWindow as Int64Window
+from zipline.lib.adjustment import Float64Multiply
+
+
+class SlidingWindow(object):
+    """
+    Wrapper around an AdjustedArrayWindow which supports monotonically
+    increasing (by datetime) requests for a sized window of data.
+
+    Parameters
+    ----------
+    window : AdjustedArrayWindow
+       Window of pricing data with prefetched values beyond the current
+       simulation dt.
+    cal_start : int
+       Index in the overall calendar at which the window starts.
+    cal_end : int
+       Index in the overall calendar at which the window
+    """
+
+    def __init__(self, window, cal_start, cal_end):
+        self.window = window
+        self.cal_start = cal_start
+        self.cal_end = cal_end
+        self.current = around(next(window), 3)
+        self.most_recent_ix = self.cal_start
+
+    def get(self, end_ix):
+        """
+        Returns
+        -------
+        out : A np.ndarray of the equity pricing up to end_ix after adjustments
+              and rounding have been applied.
+        """
+        if end_ix < self.most_recent_ix:
+            raise Exception('Can not access data after window has passed.')
+
+        anchor = end_ix - self.cal_start + 1
+        while self.window.anchor < anchor:
+            self.current = around(next(self.window), 3)
+
+        self.most_recent_ix = end_ix
+        return self.current
+
+
+class USEquityHistoryLoader(object):
+    """
+    Loader for sliding history windows of adjusted US Equity Pricing data.
+
+    Parameters
+    ----------
+    daily_reader : DailyBarReader
+        Reader for daily bars.
+    adjustment_reader : SQLiteAdjustmentReader
+        Reader for adjustment data.
+    """
+
+    def __init__(self, daily_reader, adjustment_reader):
+        self._daily_reader = daily_reader
+        self._calendar = daily_reader._calendar
+        self._adjustments_reader = adjustment_reader
+        self._daily_window_blocks = {}
+
+        self._prefetch_length = 40
+
+    def _get_adjustments_in_range(self, assets, days, field):
```

Comment:
Also, the `date_loc` needs to be `effective_date` - 1, instead of the `effective_date`.

Though, whether or not to do that shift on the date, could be parameterized.